packages/block-library/src/query-title/edit.js

```diff
@@ -48,6 +49,21 @@ export default function QueryTitleEdit( {
 		titleElement = (
 			<TagName { ...blockProps }>{ __( 'Archive title' ) }</TagName>
 		);
+	} else {
+		titleElement = (
+			<div { ...blockProps }>
+				<RichText
```

Comment:
Is there a reason to use RichText here rather than an H* element like we do for the Archive title?

Reply:
I'm not sure actually, I pulled this from the previous work on the search title in #27989. I'm guessing the intent was to use the additional functionality you get from RichText, but using the H* / TagName would be easier to read and maintain because there are fewer props. I'll try it out with just the H* tag.

Reply:
`RichText` should be used if text is meant to be editable